src/main/java/appeng/client/gui/implementations/GuiCraftingCPU.java

```diff
@@ -192,18 +192,18 @@ public void drawScreen( int mouseX, int mouseY, float btn )
 	@Override
 	public void drawFG( int offsetX, int offsetY, int mouseX, int mouseY )
 	{
-		ReadableNumberConverter converter = ReadableNumberConverter.INSTANCE;
+		final ReadableNumberConverter converter = ReadableNumberConverter.INSTANCE;
+		String title = this.getGuiDisplayName( GuiText.CraftingStatus.getLocal() );
 
-		this.fontRendererObj.drawString( this.getGuiDisplayName( GuiText.CraftingStatus.getLocal() ), 8, 7, TEXT_COLOR );
-
-		if( this.craftingCpu.eta > 0 )
+		if( this.craftingCpu.eta > 0 && !this.visual.isEmpty() )
 		{
-			final String etaText = this.getGuiDisplayName( GuiText.ETA.getLocal() );
 			final long etaInMilliseconds = TimeUnit.MILLISECONDS.convert( this.craftingCpu.eta, TimeUnit.NANOSECONDS );
-			final String etaTimeText = DurationFormatUtils.formatDuration( etaInMilliseconds, this.getGuiDisplayName( GuiText.ETAFormat.getLocal() ) );
-			this.fontRendererObj.drawString( String.format( etaText, etaTimeText ), ETA_LEFT_OFFSET, ETA_TOP_OFFSET, TEXT_COLOR );
+			final String etaTimeText = DurationFormatUtils.formatDuration( etaInMilliseconds, GuiText.ETAFormat.getLocal() );
+			title += " - " + etaTimeText;
```

Comment:
`ETA_LEFT_OFFSET` and `ETA_TOP_OFFSET` are unused now:

* either repurpose / rename them for title
* or delete them